oidc-controller/api/templates/verified_credentials.html

```diff
@@ -475,16 +482,41 @@ <h5 v-if="state.showScanned" class="fw-bolder mb-3">
         this.socket = io(location.host, {
           path: "/ws/socket.io",
           autoConnect: false,
+          reconnection: true,
+          reconnectionDelay: 1000, // How long to wait before attempting a new reconnection (in milliseconds)
+          reconnectionAttempts: Infinity, // How many reconnection attempts to make before giving up
         });
 
         this.socket.on("connect", () => {
+          console.log(
+            `Socket connecting. SID: ${this.socket.id}. PID: {{pid}}. Recovered? ${this.socket.recovered} `
+          );
           this.socket.emit("initialize", { pid: "{{pid}}" });
         });
 
+        this.socket.on("connect_error", (error) => {
+          console.error(
+            `Socket connection error: ${error}. Active: ${this.socket.active}.`
+          );
+        });
+
+        this.socket.on("disconnect", (reason, details) => {
+          console.error(
+            `Socket disconnected. Reason: ${reason}. Active: ${this.socket.active}. SID: ${this.socket.id}. PID: {{pid}}`
+          );
+          if (this.socket.active) {
+            console.log("Active connection, should auto-reconnect");
+            // temporary disconnection, the socket will automatically try to reconnect
+          } else {
+            // The connection was forcefully closed by the server or the client itself
+            // in that case, `socket.connect()` must be manually called in order to reconnect
+            this.socket.connect();
+          }
+        });
+
```

Comment:
Fairly verbose handling, might pare this down in future releases if websockets aren't having issues. Or could keep as useful debugging (this is where some enterprise-y client side error tracking like Sentry or something could be integrated as well)